model/event/DeliveryCreatedEvent.php

```diff
@@ -82,19 +84,27 @@ public function getWebhookEventName()
 
     /**
      * @return array
-     * @throws core_kernel_persistence_Exception
      */
     public function serializeForWebhook()
     {
-        $testProperty = new \core_kernel_classes_Property(DeliveryAssemblyService::PROPERTY_ORIGIN);
         return [
             'deliveryId' => $this->deliveryUri,
-            'testId' => $this->delivery->getOnePropertyValue($testProperty)->getUri(),
+            'testId' => $this->getOriginTestUri(),
         ];
     }
 
-    public function getOriginTest(): ?core_kernel_classes_Resource
+    /**
+     * @return string|null
+     */
+    public function getOriginTestUri(): ?string
     {
-        return $this->originTest;
+        if ($this->originTestUri === null) {
```

Comment:
Why is there a need to have that property's value defined externally via the constructor then?
Do we want it to be overridable?

Reply:
This logic was already [there](https://github.com/oat-sa/extension-tao-delivery-rdf/pull/345/files#diff-236c31dbbffa3833f834b0a3eedca3dcL92). In this  PR I extracted it to a separate method to be able to  reuse and left this logic here for backward compatibility to cover cases when DeliveryCreatedEvent is still triggered without test URI. Would you suggest to remove it and return `null` if test URI was not passed in constructor?

Reply:
No-no, I mean, why do we even need to pass it to constructor then?

Reply:
To avoid unnecessary DB call in case it was passed as constructor parameter.